Core/HLE/sceNet.cpp

```diff
@@ -1219,21 +1226,57 @@ static int sceNetApctlGetBSSDescEntryUser(int entryId, int infoId, u32 resultAdd
 	if (!Memory::IsValidAddress(resultAddr))
 		hleLogError(SCENET, -1, "apctl invalid arg");
 
+	// Generate an SSID name
+	char dummySSID[APCTL_SSID_MAXLEN] = "WifiAP0";
+	dummySSID[6] += static_cast<char>(entryId);
+
 	switch (infoId) {
 	case PSP_NET_APCTL_DESC_IBSS: // IBSS, 6 bytes
-		Memory::WriteStruct(resultAddr, &netApctlInfo.bssid);
+		if (entryId == 0)
+			Memory::WriteStruct(resultAddr, &netApctlInfo.bssid);
+		else {
+			// Generate a BSSID/MAC address
+			char dummyMAC[ETHER_ADDR_LEN];
+			memset(&dummyMAC, entryId, sizeof(dummyMAC));
+			// Making sure the 1st 2-bits on the 1st byte of OUI are zero to prevent issue with some games (ie. Gran Turismo)
+			dummyMAC[0] &= 0xfc;
+			Memory::WriteStruct(resultAddr, &dummyMAC);
+		}
 		break;
 	case PSP_NET_APCTL_DESC_SSID_NAME:
 		// Return 32 bytes
-		Memory::WriteStruct(resultAddr, &netApctlInfo.ssid);
+		if (entryId == 0)
+			Memory::WriteStruct(resultAddr, &netApctlInfo.ssid);
+		else {
+			Memory::WriteStruct(resultAddr, &dummySSID);
+		}
 		break;
 	case PSP_NET_APCTL_DESC_SSID_NAME_LENGTH:
 		// Return one 32-bit value
-		Memory::WriteStruct(resultAddr, &netApctlInfo.ssidLength);
+		if (entryId == 0)
+			Memory::WriteStruct(resultAddr, &netApctlInfo.ssidLength);
+		else {
+			// Calculate the SSID length
+			Memory::Write_U32((u32)strlen(dummySSID), resultAddr);
+		}
+		break;
+	case PSP_NET_APCTL_DESC_CHANNEL:
+		// FIXME: Return one 1 byte value or may be 32-bit if it this is not channel?
+		if (entryId == 0)
+			Memory::WriteStruct(resultAddr, &netApctlInfo.channel);
+		else {
+			// Generate channel for testing purposes, not even sure whether this is channel or not, MGS:PW seems to treat the data as u8
+			Memory::Write_U8(entryId, resultAddr);
+		}
 		break;
 	case PSP_NET_APCTL_DESC_SIGNAL_STRENGTH:
 		// Return 1 byte
-		Memory::WriteStruct(resultAddr, &netApctlInfo.strength);
+		if (entryId == 0)
+			Memory::WriteStruct(resultAddr, &netApctlInfo.strength);
+		else {
+			// Randomize signal strength between 1%~99% since games like MGS:PW are using signal strength to determine the strength of the recruit
+			Memory::Write_U8(rand()*99 + 1, resultAddr);
```

Comment:
rand()*99+1 will not do what you think. rand() returns very large integer numbers, so the only thing limiting here is that you just write a byte.

Probably not worth messing with a better random generator here, so the "right" way to do this with rand() should be something like:

```(int)(((float)rand()/(float)RAND_MAX)) * 99.0 + 1.0)```

Reply:
Ah you i right, i was taking the code example from here https://www.cplusplus.com/reference/cstdlib/rand/
And it should be rand() % 99 instead of rand()*99 >.< i must have been sleepy

Reply:
yeah, that'd also be acceptable for this use case.